Adds changes for rsl_rl 3.0.1

ClemensSchwarke opened the pull request:
# Description

This PR adds the necessary changes to work with the new version of RSL RL.

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [ ] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

ClemensSchwarke commented:
> Thanks for the changes, this is awesome! Looks like it could have some breaking changes for existing cfgs, should we include a check for the RSL RL version to force a version update if users have older versions installed?

Ah yes that would make a lot of sense I think. We could alternatively just keep all the configs unmodified as the new rsl rl version is backwards compatible, but I thought it would be nicer to show the updated way of handling normalization in the example configs. Any opinions on this?

kellyguo11 commented:
> > Thanks for the changes, this is awesome! Looks like it could have some breaking changes for existing cfgs, should we include a check for the RSL RL version to force a version update if users have older versions installed?
> 
> Ah yes that would make a lot of sense I think. We could alternatively just keep all the configs unmodified as the new rsl rl version is backwards compatible, but I thought it would be nicer to show the updated way of handling normalization in the example configs. Any opinions on this?

I think showing the new features would be nice as well. @Mayankm96 @ooctipus @pascal-roth @jtigue-bdai what are your thoughts?

pascal-roth commented on source/isaaclab_rl/isaaclab_rl/rsl_rl/vecenv_wrapper.py:
```diff
@@ -162,30 +136,28 @@ def episode_length_buf(self, value: torch.Tensor):
     def seed(self, seed: int = -1) -> int:  # noqa: D102
         return self.unwrapped.seed(seed)
 
-    def reset(self) -> tuple[torch.Tensor, dict]:  # noqa: D102
```
Whats the reason that the reset function is no longer needed?

ClemensSchwarke replied:
It was never used/needed.

pascal-roth commented on source/isaaclab_tasks/isaaclab_tasks/direct/allegro_hand/agents/rsl_rl_ppo_cfg.py:
```diff
@@ -14,9 +14,10 @@ class AllegroHandPPORunnerCfg(RslRlOnPolicyRunnerCfg):
     max_iterations = 10000
     save_interval = 250
     experiment_name = "allegro_hand"
-    empirical_normalization = True
```
It is not backwards compatible here, the key is missing in the new confit class (the key is missing in the config)

pascal-roth reviewed (changes requested):
Thanks for pushing that out. 

I agree we should update to the new normalisation definition but currently it is not backward compatible, should have that fixed

ClemensSchwarke commented:
> Thanks for pushing that out.
> 
> I agree we should update to the new normalisation definition but currently it is not backward compatible, should have that fixed

@pascal-roth So you are saying forcing the new rsl rl version is not the way to go?

ooctipus reviewed (approved):
I have tested locally myself, looks pretty good!

ooctipus commented:
I looked into the test a bit,

the test `test_random_actions` fails because it seems like wrapper has removed the `reset`. While this doesn't introduce any error in pure rsl-rl training pipeline this might break some user's workflow if they customized older rsl_rl and depended on the reset to construct their own environment.

The test is relying on `obs, extras = wrapper.reset` to test both observation and extra. 
But in newer wrapper code, you can't get observation and extra at the same time in any alternative. while you can get observation from `wrapper.get_observation`, extra is only accessible by calling `wrapper.step`. (Maybe we should call step rather than reset?)

I am unsure what the proper fix should be, probably rsl team has better idea, hope this provides some quick preview of source of test failure :))

ooctipus commented:
I think keeping `reset` or not keeping `reset` all make sense. If we do not keep `reset`, then maybe can you help on fixing the test case so it is testing the new rsl-rl without `reset` instead testing the old one : ))

ClemensSchwarke commented:
I added the reset function again to the wrapper but not to rsl_rl, this way rsl_rl stays cleaner but the functionality is there. I also fixed the test to work with TensorDict. Let me know if that makes sense :)

kellyguo11 commented:
hmm not quite sure why but CI doesn't seem to like this change...something is causing `ModuleNotFoundError: No module named 'pip._vendor.packaging._structures'` from pip

ClemensSchwarke commented:
I tried to debug the failing CI, but I can't really make the connection to this PR. It seems like pip is breaking because the packaging package is uninstalled, due to skrl requiring a newer version. But this would not be related to rsl_rl so maybe not the right direction :/

kellyguo11 commented:
yup quite strange. @ooctipus @ClemensSchwarke could one of you try to run the same pipeline from docker similar to what CI is doing and see if you can repro it locally?

ClemensSchwarke commented:
The test passes for me without any issues. What is failing for you @ooctipus ?

```
source/isaaclab_tasks/test/benchmarking/test_environments_training.py .                                                                                                                     [100%]{
  "rsl_rl:Isaac-Velocity-Rough-Anymal-C-v0": {
    "success": true,
    "msg": "",
    "reward": 15.3572,
    "reward_normalized": 2.1938857142857144,
    "reward_threshold": 7,
    "episode_length": 963.23,
    "episode_length_threshold": 900,
    "duration": 542.6443,
    "duration_threshold": 1800,
    "max_iterations": 300
  },
  "overall": {
    "totals": {
      "rsl_rl": 1
    },
    "successes": {
      "rsl_rl": 1
    },
    "failures_did_not_finish": {
      "rsl_rl": 0
    },
    "failures_did_not_pass_thresholds": {
      "rsl_rl": 0
    },
    "timestamp": "2025-08-25T11:41:08.402576",
    "tag": ""
  }
}


================================================================================== 1 passed in 608.53s (0:10:08) ==================================================================================
```
